docs/structure-of-a-contract.rst

```diff
@@ -91,6 +151,47 @@ Functions may be called internally or externally depending on their :ref:`visibi
 
 See the :ref:`Functions <control-structures-functions>` documentation for more information.
 
+.. _modules:
+
+Modules
+==========
+
+A module is a set of function definitions and variable declarations which enables code reuse. Vyper favors code reuse through composition, rather than inheritance.
+
+Broadly speaking, a module contains:
+
+* function definitions
+* variable declarations
+* type definitions
+
+Therefore, a module encapsulates
+
+* functionality (types and functions)
+* and state (variables)
```

Comment:
```suggestion
* functionality (types and functions); and
* state (variables)
```
suggestion, feel free to disregard

Reply:
we don't need the `and` at all imo

Reply:
i settled on https://github.com/vyperlang/vyper/pull/3947/commits/77812bb025534de3552c5a958f60dd5ff9d84513